src/installer/pkg/sfx/Microsoft.NETCore.App/Microsoft.NETCore.App.Crossgen2.sfxproj

```diff
@@ -75,20 +75,29 @@
   <Import Project="Sdk.targets" Sdk="Microsoft.DotNet.SharedFramework.Sdk" />
 
   <Target Name="ResolveReadyToRunCompilers" DependsOnTargets="ResolveRuntimeFilesFromLocalBuild">
-    <PropertyGroup> <!-- Can be removed once runtime repo switches over to 6.0 drop of the SDK repo -->
+    <!-- The following property groups can be simplified once runtime repo switches over to SDK 6.0 drop -->
+    <PropertyGroup Condition="'$(BuildArchitecture)' != '$(TargetArchitecture)'">
+      <CrossDir Condition="'$(TargetArchitecture)' == 'arm' or '$(TargetArchitecture)' == 'arm64'">x64</CrossDir>
+      <CrossDir Condition="'$(TargetArchitecture)' == 'x86'">$(BuildArchitecture)</CrossDir>
```

Comment:
What is different for x86 target architecture?

Reply:
https://github.com/dotnet/runtime/issues/47388

Reply:
And what value do you expect `$(BuildArchitecture)` to be equal to for x86?

Reply:
I believe that BuildArchitecture is the "natural" architecture of the build machine, on my box it returns "x64" and that works fine.

Reply:
It's basically an equivalent of what we have in the SPC crossgenning script,

https://github.com/dotnet/runtime/blob/9f8aab73d93156933ae65a476204bf62c02f6537/src/coreclr/crossgen-corelib.proj#L34

Reply:
Right, it will be x64. Then back to my question: why do we treat x86 differently here?  We could merge those two lines and treat all architectures (arm, arm64, x86) uniformly.

Reply:
Hmm, I was under the impression that it would break the build on a machine that's actually x86 but I guess that runtime repo may have trouble building there anyway.

Reply:
These lines are under `Condition="'$(BuildArchitecture)' != '$(TargetArchitecture)'"`.  So if `TargetArchitecture` is x86, then `BuildArchitecture` cannot be x86.

Reply:
OK, that's true, I'll change it as you suggest then, thanks for pointing that out.

Reply:
Fixed in 6th commit.

Reply:
Thank you!